tests/test_cli_compile.py

```diff
@@ -69,26 +73,26 @@ def test_command_line_setuptools_read(pip_conf, runner):
         (["setup.py", "--output-file", "output.txt"], "output.txt"),
     ],
 )
-def test_command_line_setuptools_output_file(pip_conf, options, expected_output_file):
+def test_command_line_setuptools_output_file(
+    pip_conf, runner, options, expected_output_file
+):
     """
     Test the output files for setup.py as a requirement file.
     """
-    runner = CliRunner(mix_stderr=False)
-    with runner.isolated_filesystem():
-        package = open("setup.py", "w")
-        package.write(
-            dedent(
-                """\
-                from setuptools import setup
-                setup(install_requires=[])
-                """
-            )
+    package = open("setup.py", "w")
+    package.write(
+        dedent(
+            """\
+            from setuptools import setup
+            setup(install_requires=[])
+            """
         )
-        package.close()
+    )
+    package.close()
```

Comment:
Thanks for refactoring this test! Just one note: could you also refactor-out `package = open(...)` thing? For example:

```python
with open(...) as package:
   ...
```